src/HttpLib/protocol/UDP/PseudoTCPMessage.java

```diff
@@ -53,7 +60,39 @@ private void createPackets() {
         }
     }
 
+    public void addPacket(PseudoTCPPacket packet, int index) {
+        growPackets(index);
+        _packets.set(index, packet);
+    }
+
+    private void growPackets(int index) {
+        _packets.ensureCapacity(index);
+        while (_packets.size() <= index)
+            _packets.add(new PseudoTCPPacket());
+
```

Comment:
I'm curious, this is necessary even with ensureCapacity ??

Reply:
ensureCapacity removed, originally thought it would auto grow the ArrayList